test/e2e/orderRefunds.test.ts

```diff
@@ -0,0 +1,169 @@
+      await token.mint(settlement.address, ethers.utils.parseEther("1.0"));
+      for (const trader of traders.slice(0, 2)) {
+        await token.mint(trader.address, ethers.utils.parseEther("1000000.0"));
+        await token
+          .connect(trader)
+          .approve(allowanceManager.address, ethers.constants.MaxUint256);
+      }
+    }
+
+    const ORDER_VALIDITY = 60; // seconds
+    const prepareBatch = async () => {
+      const { timestamp } = await ethers.provider.getBlock("latest");
+      const validTo = timestamp + ORDER_VALIDITY;
+
+      const sellOrder = {
+        kind: OrderKind.SELL,
+        partiallyFillable: false,
+        sellToken: owl.address,
+        buyToken: dai.address,
+        sellAmount: ethers.utils.parseEther("100.0"),
+        buyAmount: ethers.utils.parseEther("100.0"),
+        feeAmount: ethers.utils.parseEther("0.1"),
+        validTo,
+        appData: 1,
+      };
+
+      const buyOrder = {
+        kind: OrderKind.BUY,
+        partiallyFillable: false,
+        buyToken: owl.address,
+        sellToken: dai.address,
+        buyAmount: ethers.utils.parseEther("100.0"),
+        sellAmount: ethers.utils.parseEther("110.0"),
+        feeAmount: ethers.utils.parseEther("0.1"),
+        validTo,
+        appData: 2,
+      };
+
+      const encoder = new SettlementEncoder(domainSeparator);
+      await encoder.signEncodeTrade(
+        sellOrder,
+        traders[0],
+        SigningScheme.TYPED_DATA,
+      );
+      await encoder.signEncodeTrade(
+        buyOrder,
+        traders[1],
+        SigningScheme.TYPED_DATA,
+      );
+
+      const orderUids = [
+        computeOrderUid({
+          orderDigest: hashOrder(sellOrder),
+          owner: traders[0].address,
+          validTo,
+        }),
+        computeOrderUid({
+          orderDigest: hashOrder(buyOrder),
+          owner: traders[1].address,
+          validTo,
+        }),
+      ];
+
+      return [encoder, orderUids] as const;
+    };
+
+    const [encoder1, orderUids] = await prepareBatch();
+    const txWithoutRefunds = await settlement.connect(solver).settle(
+      encoder1.tokens,
+      encoder1.clearingPrices({
+        [owl.address]: ethers.utils.parseEther("1.05"),
+        [dai.address]: ethers.utils.parseEther("1.0"),
+      }),
+      encoder1.encodedTrades,
+      encoder1.encodedInteractions,
+      "0x",
+    );
+    const { gasUsed: gasUsedWithoutRefunds } = await txWithoutRefunds.wait();
+
+    await ethers.provider.send("evm_increaseTime", [ORDER_VALIDITY + 1]);
+    await ethers.provider.send("evm_mine", []);
+
+    const [encoder2] = await prepareBatch();
+    const txWithRefunds = await settlement.connect(solver).settle(
+      encoder2.tokens,
+      encoder2.clearingPrices({
+        [owl.address]: ethers.utils.parseEther("1.05"),
+        [dai.address]: ethers.utils.parseEther("1.0"),
+      }),
+      encoder2.encodedTrades,
+      encoder2.encodedInteractions,
+      encodeOrderRefunds(orderUids),
```

Comment:
What about reworking the encoder so that this part reads like this:
```suggestion
      encoder2.encodedTrades,
      encoder2.encodedInteractions,
      encoder2.encodedRefunds,
```
In principle, refunds have nothing special compared to interactions or refunds.

Reply:
Thats a very nice suggestion.

Reply:
Implemented!